[Darwin] New XPC Service

kiel-apple opened the pull request:
lighter / more direct XPC wrappers for MTRDevice and MTRDeviceController

kiel-apple committed:
empty xpc service files

kiel-apple committed:
XPC service needs to be C++ to use MTRLogging

kiel-apple committed:
add tests; stub method signature

jtung-apple commented on src/darwin/Framework/CHIP/MTRXPCService.h:
```diff
@@ -0,0 +1,27 @@
+//
```
Nit: please remove this line, and the same for the other files

kiel-apple committed:
start configuring test XPC listener

kiel-apple committed:
add first test method to XPC service protocol

kiel-apple committed:
tests: remove peformance test; add headers

kiel-apple committed:
make MTRXPCService testable

kiel-apple committed:
more working lines of XPC listener setup

kiel-apple committed:
add dummy XPC service for testing

kiel-apple committed:
add replying method to service protocol / implementation

kiel-apple committed:
remove log - reply contents can be asserted

kiel-apple committed:
make test interesting

kiel-apple committed:
linty formatty

kiel-apple committed:
remove personal development team from pbxproj

kiel-apple committed:
fix new file headers

kiel-apple committed:
less whitespace

restyled-commits committed:
Restyled by whitespace

restyled-commits committed:
Restyled by clang-format

kiel-apple committed:
add new XPC device controller

kiel-apple committed:
sketch out a first XPC method

kiel-apple commented on src/darwin/Framework/CHIPTests/TestHelpers/MTRTestServerAppRunner.m:
```diff
@@ -21,11 +21,13 @@
 
 static unsigned sAppRunnerIndex = 1;
 
+#if HAVE_NSTASK
 // kBasePort gets the discriminator added to it to figure out the port the app
 // should be using.  This ensures that apps with distinct discriminators use
 // distinct ports.
 static const uint16_t kMinDiscriminator = 1111;
 static const uint16_t kBasePort = 5542 - kMinDiscriminator;
+#endif // HAVE_NSTASK
```
#34970

kiel-apple committed:
store WIP for rebase

kiel-apple committed:
fix pbxproj merge mistake

kiel-apple committed:
make WIP slightly more sensible

kiel-apple committed:
WIP: remote proxy obj in `MTRDeviceController_XPC`

kiel-apple committed:
more experiments

kiel-apple committed:
WIP:  return type issue

run `MTRXPCServiceTests testMTRXPCServiceSetup` and see selection of proxy object method in `MTRDeviceController_XPC.m`/`initWithXPCListenerEndpointForTesting` around line 44 to see the problem.

kiel-apple committed:
add `MTRDevice_XPC`

nothing exciting yet

kiel-apple committed:
add device controller over XPC parameters stub

kiel-apple committed:
add init split for XPC device controllers

not yet implemented, but there

woody-apple commented on src/darwin/Framework/CHIP/MTRDeviceControllerOverXPCParameters.h:
```diff
@@ -0,0 +1,25 @@
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MTRDeviceControllerOverXPCParameters : MTRDeviceControllerAbstractParameters
```
Let's come up with a better name for this, I wish I had a better suggestion, but Over is not great. Maybe MTRDeviceControllerXPCParameters?

kiel-apple replied:
Yep, unattached to the current name so that's as good as any in my view.  Making the change.

kiel-apple committed:
rename new XPC parameters class

`OverXPC` is already taken / means something else

restyled-commits committed:
Restyled by whitespace

restyled-commits committed:
Restyled by clang-format

kiel-apple committed:
remove failing exploratory test

kiel-apple committed:
lintfix

woody-apple committed:
Examples

kiel-apple committed:
quickly attempt to fix test

unused variable -> warning -> error

restyled-commits committed:
Restyled by whitespace

restyled-commits committed:
Restyled by clang-format

kiel-apple committed:
inevitably the file will need to be obj-C++

kiel-apple committed:
prepare to use mach service name to create XPC connection

kiel-apple committed:
allow for other XPC controller connection types for now

once mach service is working, i don't expect to need others, but easier to collapse classes than raise.

kiel-apple committed:
constant for well-known matter plugin service name

kiel-apple committed:
note to self and re-ordering of this code now what `XPCParameters` is a subclass

kiel-apple committed:
XPC parameters should support secure coding

kiel-apple committed:
don't keep around object proxy; make a new one when needed.  try calling the `checkIn` method when connecting.  remove old test methods that served their purpose.

anush-apple commented on src/darwin/Framework/CHIP/MTRDeviceControllerXPCParameters.h:
```diff
@@ -0,0 +1,33 @@
+
+NS_ASSUME_NONNULL_BEGIN
+
+static NSString * MatterPluginWellKnownServiceName = @"com.apple.home.matter.framework.xpc";
```
Why is this here?

kiel-apple replied:
removed

kiel-apple committed:
that doesn't need to be there.

restyled-commits committed:
Restyled by whitespace

restyled-commits committed:
Restyled by clang-format

kiel-apple committed:
no longer fighting with compiler, have some `instancetype`

kiel-apple committed:
extremely normal object construction

kiel-apple committed:
somehow this header got thrown out

restyled-commits committed:
Restyled by whitespace

restyled-commits committed:
Restyled by clang-format

woody-apple committed:
Updating XPC interfaces

kiel-apple committed:
hide `initWithMachServiceName:options:` constructor

tvOS/iOS compilation issues need a look

kiel-apple committed:
make `MTRDeviceController_XPC` an XPC client

via `MTRXPCClientProtocol_MTRDeviceController`, which for the moment is empty

kiel-apple committed:
remove initial tests

they served their purpose well but are no longer relevant

kiel-apple committed:
call mach service constructor when appropriate

also logging

kiel-apple committed:
remove some obsolete test methods

kiel-apple committed:
more obsoletes fixed

kiel-apple committed:
more obsolete removal and logging tweaks

kiel-apple committed:
buildability tweaks

woody-apple committed:
Moving along

woody-apple committed:
Fixing header

woody-apple committed:
Fixing macros and codable

woody-apple committed:
Adding invoke?

restyled-commits committed:
Restyled by whitespace

restyled-commits committed:
Restyled by clang-format

woody-apple committed:
Actually hooking up to XPCConnection

woody-apple committed:
Actually hooking up to XPCConnection

woody-apple committed:
Adding XPC Parameters

woody-apple committed:
Adding XPC Parameter hookup

restyled-commits committed:
Restyled by whitespace

restyled-commits committed:
Restyled by clang-format

woody-apple committed:
true => YES
